main.tf

```diff
@@ -11,9 +11,14 @@ locals {
 
   subnet_ids_provided           = var.subnet_ids != null && length(var.subnet_ids) > 0
   db_subnet_group_name_provided = var.db_subnet_group_name != null && var.db_subnet_group_name != ""
+  is_replica                    = try(length(var.replicate_source_db), 0) > 0
```

Comment:
This may cause an issue if a replicate_source_db is passed in from another module and has not been created yet. Meaning the replicate source db would have to be terraform apply first and then you could apply the full module.

When we have run into this in the past, we either need to check if the input is not null or turn the input var into a list(string) and make it one of those 0 or 1 only lists.

Reply:
@nitrocode references to resources created during apply are only a problem when they affect the Terraform plan, specifically the address of created resources. Is `is_replica` used in `count` or `for_each` somewhere? I did not see it.

Reply:
Thanks for confirming. I'll make note of that in the future.